pkg/api/customization/nodepool/validator.go

```diff
@@ -2,22 +2,51 @@ package nodepool
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/rancher/norman/api/access"
 	"github.com/rancher/norman/httperror"
 	"github.com/rancher/norman/types"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
 	mgmtSchema "github.com/rancher/types/apis/management.cattle.io/v3/schema"
 	mgmtclient "github.com/rancher/types/client/management/v3"
 )
 
-func Validator(request *types.APIContext, schema *types.Schema, data map[string]interface{}) error {
-	nodetemplateID, _ := data["nodeTemplateId"].(string)
-	if err := access.ByID(request, &mgmtSchema.Version, mgmtclient.NodeTemplateType, nodetemplateID, nil); err != nil {
+type Validator struct {
+	NodePoolLister v3.NodePoolLister
+}
+
+func (v *Validator) Validator(request *types.APIContext, schema *types.Schema, data map[string]interface{}) error {
+	// does we have access to the nodepool?
+	if err := access.ByID(request, &mgmtSchema.Version, mgmtclient.NodePoolType, request.ID, nil); err != nil {
 		if httperror.IsNotFound(err) || httperror.IsForbidden(err) {
-			return httperror.NewAPIError(httperror.NotFound, fmt.Sprintf("unable to find node template [%s]", nodetemplateID))
+			return httperror.NewAPIError(httperror.NotFound, fmt.Sprintf("unable to access node pool [%s]", request.ID))
 		}
 		return httperror.NewAPIError(httperror.ServerError, err.Error())
+	}
+
+	//check if nodetemplate changing
+	split := strings.SplitN(request.ID, ":", 2)
+	cluster, nodepool := split[0], split[1]
 
+	np, err := v.NodePoolLister.Get(cluster, nodepool)
+	if err != nil {
+		return err
 	}
+
+	nodetemplateID, _ := data["nodeTemplateId"].(string)
```

Comment:
I think it's possible for nodeTemplate to not be in the request if its an update. This should also be allowed